modules/unsupported/mbstyle/src/test/java/org/geotools/mbstyle/expression/MBLookupTest.java

```diff
@@ -0,0 +1,130 @@
+        } catch(Exception e) { }
+        // test nested "get expression
+        final JSONObject jn = getObjectByLayerId("getExpression", "paint");
+        Object on = getExpressionEvaluation(jn, "halo-color");
+        assertEquals("RED", on);
+    }
+
+    /**
+     * Verify that an "at" lookup expression can be parsed correctly.
+     */
+    @Test
+    public void testParseAtExpression() {
+        // test "at" expression for array
+        final JSONObject j = getObjectByLayerId("atExpression", "paint");
+        Object o = getExpressionEvaluation(j, "text-color");
+        assertEquals("#006fde", o);
+        // test nested "at" expression
+        final JSONObject jn = getObjectByLayerId("atExpression", "paint");
+        Object on = getExpressionEvaluation(j, "halo-color");
+        assertEquals("length", on);
+    }
+
+    /**
+     * Verify that a "length" lookup expression can be parsed correctly.
+     */
+    @Test
+    public void testParseLengthExpression() {
+        // test string length
+        final JSONObject j = getObjectByLayerId("lengthExpression", "paint");
+        Object o = getExpressionEvaluation(j, "text-color");
+        assertEquals(9, o);
+        // test array length
+        final JSONObject jl = getObjectByLayerId("lengthExpression", "layout");
+        Object ol = getExpressionEvaluation(jl, "text-field");
+        assertEquals(9, ol);
+        // test nested length expression
+        final JSONObject jn = getObjectByLayerId("lengthExpression", "paint");
+        Object on = getExpressionEvaluation(j, "halo-color");
+        assertEquals(9, on);
+    }
+
+    /**
+     * Verify that a "has" Lookup expression can be parsed correctly.
+     */
+    @Test
+    public void testParseHasExpression() {
+        //
+        final JSONObject j = getObjectByLayerId("hasExpression", "paint");
+        Object o = getExpressionEvaluation(j, "text-color");
+        assertEquals(true, o);
+        // "has" expression test
+        final JSONObject jh = getObjectByLayerId("hasExpression", "layout");
+        assertNotNull(jh);
+        Object oh = getExpressionEvaluation(jh, "text-field");
+        assertEquals(true, oh);
+        // nested "has" expression test
+        final JSONObject jn = getObjectByLayerId("hasExpression", "paint");
+        Object on = getExpressionEvaluation(jn, "halo-color");
+        assertEquals(true, on);
+        // Verify that a "has" lookup expression transforms correctly to SLD.
+        MBStyle getTest = MBStyle.create(mbstyle);
+        SymbolMBLayer rgbLayer = (SymbolMBLayer) getTest.layer("hasExpression");
+        List<FeatureTypeStyle> getFeatures = rgbLayer.transformInternal(getTest);
+        try {
+            String xml = new SLDTransformer().transform(getFeatures.get(0));
+            assertTrue(xml.contains("<ogc:Function name=\"PropertyExists\"><ogc:Literal>name</ogc:Literal></ogc:Function>"));
+        } catch(Exception e) { }
```

Comment:
Unless this test expects to throw an Exception here, I think it's better to declare the test method `throws Exception` and don't bother with the try-catch here.